### ISSUE TITLE
getSpanLImits method missing in Tracer class

### ISSUE THREAD

**Issue by sajitron** [DataDog/dd-trace-js#4313]
dd-trace Tracer [class](https://github.com/DataDog/dd-trace-js/blob/79f0d64cc798fbe99d8f59028b4906b93162e53c/packages/dd-trace/src/opentelemetry/tracer.js#L11) doesn't implement the `getSpanLimits` method which is present in the opentelemetry Tracer [class](https://github.com/open-telemetry/opentelemetry-js/blob/806fa97ebd8024c5c78e818ede350c3e1874777a/packages/opentelemetry-sdk-trace-base/src/Tracer.ts#L250).

This breaks prisma instrumentation for `@prisma/instrumentation: >4.13.0` with the following error:

```
uncaughtException: parentTracer.getSpanLimits is not a function
TypeError: parentTracer.getSpanLimits is not a function
    at new Span (/usr/src/app/node_modules/@opentelemetry/sdk-trace-base/build/src/Span.js:59:41)
    at /usr/src/app/node_modules/@prisma/instrumentation/dist/chunk-VVAFFO6L.js:59:20
    at Array.forEach (<anonymous>)
    at ActiveTracingHelper.createEngineSpan (/usr/src/app/node_modules/@prisma/instrumentation/dist/chunk-VVAFFO6L.js:44:27)
    at eo.createEngineSpan (/usr/src/app/node_modules/@prisma/client/runtime/library.js:122:1645)
    at wt.logger (/usr/src/app/node_modules/@prisma/client/runtime/library.js:112:1167)
    at /usr/src/app/node_modules/@prisma/client/runtime/library.js:112:922
```

**Comment by sajitron** [comment]
> Please tell me how you solved it. I'm having the same problem.

You need to do a patch of the library.

Add the following method to the `Tracer` class within the file - `node_modules/dd-trace/packages/dd-trace/src/opentelemetry/tracer.js`:

```
getSpanLimits () {
   return {}
}

**Comment by VicenteRD** [comment]
Hello! Version 5.19.0 applied this patch, which according to @sajitron should fix this issue, but I tried it and it didn't do it. I don't understand how patching that version of the Tracer class would fix the issue since the tracer that is passed to Prisma is the one at the base of the package, isn't it?
Am I doing something wrong here? Can someone share their setup working with version 5.19.0?
Thanks!

**Comment by VicenteRD** [comment]
I've figured it out. Turns out, when calling `register` on a `TracerProvider`, it can fail to register the provider as the global one if there's a version mismatch for the OpenTelemetry API package, so you need to make sure that, if you're importing the API yourself, you do so with the same version the `dd-trace` package does. In this case, 1.8.0, with 1.9.0 being the latest.

TL;DR: Make sure that _if_ you have `@opentelemetry/api` in your package.json, it's set to whatever version `dd-trace` depends on (for 5.19.0, API version is 1.8.0).

**Comment by sajitron** [comment]
Seems the bug has been fixed in this [PR](https://github.com/DataDog/dd-trace-js/pull/4454).
